Pawel - icons to buttons plus Playlist

pdob-git opened the pull request:
Icons to play, pause, etc. buttons have been added.
First version of Playlist has been created.

pdob-git committed:
Icons have been added to play buttons

Icons have been added to play buttons.
Icons folder have been added to resources.
Modifications:
	modified:       src/main/java/org/player/mp3player/controllers/MainWindow.java
	new file:       src/main/resources/org/player/mp3player/icons/eject.png
	new file:       src/main/resources/org/player/mp3player/icons/next.png
	new file:       src/main/resources/org/player/mp3player/icons/pause.png
	new file:       src/main/resources/org/player/mp3player/icons/play.png
	new file:       src/main/resources/org/player/mp3player/icons/previous.png
	new file:       src/main/resources/org/player/mp3player/icons/stop.png

pdob-git committed:
Play list added:

Play list adde. Sigleton created on Music class.
Lombok added to MusicItem
Style sheets to correct windows design
Modifications:
	modified:       pom.xml
	modified:       src/main/java/org/player/mp3player/App.java
	modified:       src/main/java/org/player/mp3player/controllers/MainWindow.java
	new file:       src/main/java/org/player/mp3player/controllers/PlayListWindow.java
	modified:       src/main/java/org/player/mp3player/model/Music.java
	new file:       src/main/java/org/player/mp3player/model/MusicItem.java
	new file:       src/main/java/org/player/mp3player/repositories/FileDataLoader.java
	modified:       src/main/java/org/player/mp3player/services/DataLoader.java
	new file:       src/main/resources/org/player/mp3player/controllers/MainWindow.css
	 renamed:       src/main/resources/org/player/mp3player/MainWindow.fxml -> src/main/resources/org/player/mp3player/controllers/MainWindow.fxml
	new file:       src/main/resources/org/player/mp3player/controllers/PlayListWindow.css
	new file:       src/main/resources/org/player/mp3player/controllers/PlayListWindow.fxml

poweredbychocolate commented on src/main/java/org/player/mp3player/App.java:
```diff
@@ -17,10 +17,11 @@ public static void main(String[] args) {
     @Override
     public void start(Stage stage) throws Exception {
-        Parent root = FXMLLoader.load(getClass().getResource("MainWindow.fxml"));
+        Parent root = FXMLLoader.load(getClass().getResource("controllers/MainWindow.fxml"));
```
Pliki fxml w katalogu źródeł raczej powinny zostać umieszczone w podkatalogu fxml, od razu będzie wiadomo że ten katalog to szablony ekranów. Tak samo css i obrazki

pdob-git replied:
Poprawione. Fxml w fxml, Css w css, obrazki w icons

poweredbychocolate commented on src/main/java/org/player/mp3player/controllers/MainWindow.java:
```diff
@@ -118,6 +160,91 @@ public void repeatPlayList(ActionEvent event) {
+
+        //Set Open Button Icon
+        imageOpen = new Image(this.getClass().getResourceAsStream("../icons/eject.png"));
+        imageViewOpen = new ImageView(imageOpen);
```
Tan kod powtarza się wielokrotnie, można go zapakować do jednej metody przekazując jako parametry przycisk, tekst podpowiedzi. obrazek albo ścieżkę obrazka albo w dwie metody które kolejno zwrócą widok obrazka i tooltip

pdob-git replied:
Kod dodawania ikon został przeniesiony do osobnej metody i zastosowany na wszystkich przyciskach

poweredbychocolate commented on src/main/java/org/player/mp3player/model/Music.java:
```diff
@@ -1,4 +1,28 @@
+    public ArrayList<MusicItem> playList;
+
+    private Music(ArrayList<MusicItem> playList) {
+        this.playList = playList;
```
Konstruktor z parametrem jak i pobranie instancji przeczy zostawaniu wzorca singleton. Singleton ma zapewniać ze uruchomiona zostanie dokładnie jedna instancja ale też że raz stworzona będzie dostępna wszędzie co w tym wypadku jest niemożliwe bez playlisty

pdob-git replied:
Zostało to zmienione. Są dwie metody `getInstance` jedna z argumentami druga bez.
Obie zapewniało, że jest tworzony tylko jeden obiekt.

pdob-git committed:
Modifications per pull request comments: Pawel - icons to buttons plus Playlist #2

Singleton music modified to ensure that Music is created only once.
FXML and CSS files location modified.
Code to create buttons icons have been modified and moved to separate method

Modifications:
	modified:       src/main/java/org/player/mp3player/App.java
	modified:       src/main/java/org/player/mp3player/controllers/MainWindow.java
	modified:       src/main/java/org/player/mp3player/model/Music.java
	modified:       src/main/java/org/player/mp3player/repositories/FileDataLoader.java
	renamed: src/main/resources/org/player/mp3player/controllers/MainWindow.css -> src/main/resources/org/player/mp3player/css/MainWindow.css
	renamed: src/main/resources/org/player/mp3player/controllers/PlayListWindow.css -> src/main/resources/org/player/mp3player/css/PlayListWindow.css
	renamed: src/main/resources/org/player/mp3player/controllers/MainWindow.fxml -> src/main/resources/org/player/mp3player/fxml/MainWindow.fxml
	renamed: src/main/resources/org/player/mp3player/controllers/PlayListWindow.fxml -> src/main/resources/org/player/mp3player/fxml/PlayListWindow.fxml
